kafka-admin/src/main/resources/graphql-schema/kafka-admin.graphql

```diff
@@ -1,11 +1,67 @@
-extend type Query {
-    topic(topicName: String): Topic
-    topicList(filter: String): [Topic]
+type TopicDescription {
+    #id: ID
+    name: String
+
+    """@transient"""
+    partitions: [Partitions]
+    config: TopicConfig
+}
+
+type Partitions {
+    partition: Int
+    replicas: [Replicas]
+}
+
+type Replicas {
+    id: String
+    inSync: Boolean
 }
 
-type Topic {
+type TopicOnlyName {
+    #id: ID
     name: String
+}
+
+type TopicConfig {
+    partitionCount: Int
+    replicationFactor: Int
     isInternal: Boolean
+    # config
+    minInsyncReplicas: Int
+    pairs: [ConfigEntry]
+}
+
+type ConfigEntry {
+    key: String
+    value: String
+}
+
+input CreateOrMutateTopicConfigInput {
+    retentionDays: Int
+    minInSyncReplicas: Int
     partitionCount: Int
     replicationFactor: Int
+    isInternal: Boolean
+}
+
+input CreateTopicInput {
+    name: String
+    config: CreateOrMutateTopicConfigInput
+}
+
+input MutateTopicInput {
+    #id: ID!
+    name: String!
+    config: CreateOrMutateTopicConfigInput
+}
+
+extend type Mutation {
+    createTopic(input: CreateTopicInput!): TopicDescription
+    updateTopic(input: MutateTopicInput!): TopicDescription
+    deleteTopic(name: String!): String
+}
+
+extend type Query {
+    topicList(filter: String): [TopicOnlyName!]
+    topicDescription(name: String!): TopicDescription!
```

Comment:
Trivial/Style

This seems to be leaking server side implementation details. 
If I want to have some information on topic I should be able to get it with the fields that I have request it and if backend needs to do 6-7 calls to do so then so be it. 
Making client doing multiple calls is more restful approach

Reply:
So you would rather see `topicDescription` returning just [isInternal, name, partitions] and another query (for example) `topicConfiguration` to get all these configs? https://kafka.apache.org/documentation/#topicconfigs

Reply:
This is comming from GrapHQL ideology that there is query that returns entire object and it's relationships and client can have freedom to pick what he wants. 

Spliting schema objects towards what client might want (list of just names or list of the topic or the configs etc.) seems to be more aligned with the way rest works.

Then client can pick query by selecting what they want:
https://i0.wp.com/css-tricks.com/wp-content/uploads/2020/01/Screen-Shot-2020-01-27-at-11.32.20.png?resize=3104%2C1978&ssl=1

Obviously this is your project and there might be different reasons why we want to split topic into 3 different objects so do not feel that I'm really suggesting something that should be done. 
More like trying to help with schema and avoiding some pitfalls early on.